src/services/offerService.js

```diff
@@ -228,14 +230,98 @@ export const newOffer = async ({
         const json = await res.json();
 
         if (!res.ok) {
+            createEvent(EVENT_TYPES.ERROR(
+                OFFER_NEW_METRIC_ID,
+                ErrorTypes.BAD_RESPONSE,
+                res.status
+            ));
+
+            throw json.errors;
+        }
+
+        createEvent(EVENT_TYPES.SUCCESS(OFFER_NEW_METRIC_ID));
+        return json;
+
+    } catch (error) {
+        createEvent(EVENT_TYPES.ERROR(
+            OFFER_NEW_METRIC_ID,
+            ErrorTypes.NETWORK_FAILURE
+        ));
+
+        if (Array.isArray(error)) throw error;
+        throw [{ msg: "Unexpected Error. Please try again later." }];
+    }
+});
+
+export const editOffer = measureTime(TIMED_ACTIONS.OFFER_EDIT, async ({
+    offerId,
+    title,
+    publishDate,
+    publishEndDate,
+    jobMinDuration,
+    jobMaxDuration,
+    jobStartDate,
+    description,
+    contacts,
+    isPaid,
+    vacancies,
+    jobType,
+    fields,
+    technologies,
+    location,
+    coordinates,
+    requirements,
+}) => {
+    const data = {
+        title,
+        publishDate,
+        publishEndDate,
+        jobMinDuration,
+        jobMaxDuration,
+        jobStartDate,
+        description,
+        contacts,
+        isPaid,
+        vacancies,
+        jobType,
+        fields,
+        technologies,
+        location,
+        coordinates,
+        requirements,
+    };
+
+    try {
+        const res = await fetch(`${API_HOSTNAME}/offers/edit/${offerId}`, {
+            method: "POST",
+            headers: {
+                "Content-Type": "application/json",
+            },
+            credentials: "include",
+            body: JSON.stringify(data),
+        });
+        const json = await res.json();
+
+        if (!res.ok) {
+            createEvent(EVENT_TYPES.ERROR(
+                OFFER_NEW_METRIC_ID,
+                ErrorTypes.BAD_RESPONSE,
+                res.status
+            ));
+
             throw json.errors;
         }
-        // TODO count metrics
+
+        createEvent(EVENT_TYPES.SUCCESS(OFFER_EDIT_METRIC_ID));
         return json;
 
     } catch (error) {
-        // TODO count metrics
+        createEvent(EVENT_TYPES.ERROR(
+            OFFER_EDIT_METRIC_ID,
+            ErrorTypes.NETWORK_FAILURE
+        ));
```

Comment:
We don't want to create a network failure event when the `!res.ok` condition triggers., hence the `isErrorRegistered` flag in other services. I think we need that here as well